src/java/org/apache/cassandra/index/sai/disk/vector/JVectorLuceneOnDiskGraph.java

```diff
@@ -62,14 +59,7 @@ protected JVectorLuceneOnDiskGraph(SegmentMetadata.ComponentMetadataMap componen
     /**
      * See CassandraDiskANN::search
      */
-    public abstract VectorPostingList search(float[] queryVector, int topK, int limit, Bits bits, QueryContext context);
-
-    /**
-     * See CassandraDiskANN::search
-     */
-    public abstract VectorPostingList search(float[] queryVector, int topK, float threshold, int limit, Bits bits, QueryContext context);
-
-    public abstract ScoredRowIdIterator searchWithScores(float[] queryVector, int topK, Bits bits, QueryContext context);
+    public abstract NodeScoreToScoredRowIdIterator search(float[] queryVector, int topK, float threshold, Bits bits, QueryContext context, IntConsumer nodesVisited);
```

Comment:
I am not a huge fan of the way the Consumer pattern makes it hard to follow the execution flow.  Can we put nodesVisited back in the return class or is there a reason that doesn't work as well anymore?

Reply:
... I guess this is because `resume` means we have multiple visit counts per iterator now.  I can't think of a better solution, so let's go with the Consumer.

Reply:
I agree that the Consumer pattern can be hard to follow. I went with this design to make it easy to process the nodesVisited when we call the `resumeSearch` class. I'll look to see if there is an easy way to refactor it out.